kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/ColumnMappingSuite.scala

```diff
@@ -201,6 +287,262 @@ class ColumnMappingSuite extends AnyFunSuite {
         (k: AnyRef) => assertThat(k).asString.startsWith("col-"))
   }
 
+  test("assign id and physical name to new table with nested struct type" +
+    " and enableIcebergCompatV2=true") {
+    val schema: StructType =
+      new StructType()
+        .add("a", StringType.STRING)
+        .add("b",
+          new StructType()
+            .add("d", IntegerType.INTEGER)
+            .add("e", IntegerType.INTEGER))
+        .add("c", IntegerType.INTEGER)
+
+    val metadata: Metadata =
+      ColumnMapping.updateColumnMappingMetadata(
+        createMetadata(schema).withNewConfiguration(
+          Maps.newHashMap(TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey, "true")), "id", true)
+
+    assertThat(metadata.getSchema.get("a").getMetadata.getEntries)
+      .containsEntry(ColumnMapping.COLUMN_MAPPING_ID_KEY, (1L).asInstanceOf[AnyRef])
+      .hasEntrySatisfying(
+        ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY,
+        (k: AnyRef) => assertThat(k).asString.startsWith("col-"))
```

Comment:
This pattern is repeated so much throughout this suite that I think it warrants taking a step back and at least

(1) thinking if we can simplify and move this to a helper function

or

(2) re-think how we are doing these tests. Have you looked at DeltaColumnMappingSuite in delta-spark. It created a `assertEqual` helpler method that takes 2 schemas. It also created a bunch of schemas at the top that could be re-used throughout tests.

Reply:
I agree and I've extracted some things out into a helper method. That should make the test more readable and it's easier to spot which IDs are being expected where